src/PageContent/Calendar/DescriptionContent.js

```diff
@@ -0,0 +1,208 @@
+import moment from 'moment';
+import { AccountType, FoodUnit } from '../../Enums';
+import './Content.css';
+import groceries from '../../icons/groceries.svg';
+import plus from '../../icons/plus-button.svg';
+
+class DescriptionContent extends Component {
+    constructor(props) {
+        super(props);
+        let donation = [];
+        if (this.props.delivery.donationDescription) {
+            donation = this.props.delivery.donationDescription;
+        }
+        this.state = {
+            edit: false,
+            donationObject: donation,
+            accountType: this.props.accountType
+        };
+
+        this.edit = this.edit.bind(this);
+        this.handleChange = this.handleChange.bind(this);
+        this.addRow = this.addRow.bind(this);
+    }
+    edit() {
+        this.setState({
+            edit: true
+        });
+    }
+
+    addRow() {
+        this.edit();
+        let blankRow = { name: '', amount: 0, unit: '' };
+        let currentDonation = this.state.donationObject;
+        currentDonation.push(blankRow);
+        this.setState({
+            donationObject: currentDonation
+        });
+    }
+
+    stringifyDonation(donation) {
+        let foodList = '';
+        if (donation !== undefined) {
+            if (donation.length > 0) {
+                foodList +=
+                    donation[0].name +
+                    ' ' +
+                    donation[0].amount +
+                    ' ' +
+                    donation[0].unit;
+                for (let i = 1; i < donation.length; i++) {
+                    foodList +=
+                        ', ' +
+                        donation[i].name +
+                        ' ' +
+                        donation[i].amount +
+                        ' ' +
+                        donation[i].unit;
+                }
+            }
+        }
+        return foodList;
+    }
+
+    handleChange(e) {
+        e.preventDefault();
+        // pass these in from firebase
+        let newDonation = [];
+        for (let i = 0; i < this.state.donationObject.length; i++) {
+            let name = e.target[i + 'name'].value;
+            let amount = e.target[i + 'amount'].value;
+            let unit = e.target[i + 'unit'].value;
+            if (name !== '') {
+                newDonation.push({ name: name, amount: amount, unit: unit });
+            }
+        }
+        this.setState({
+            donationObject: newDonation,
+            edit: false,
+            editedBy: this.props.accountOwnerName,
+            editedAt: moment().format('MM/DD h:mma')
+        });
+
+        // TODO: push data to firebase
+    }
+
+    render() {
+        let donation = this.stringifyDonation(this.state.donationObject);
+
+        let editDonation = '';
+
+        if (this.state.donationObject.length > 0) {
+            editDonation = this.state.donationObject.map((item, index) => {
+                return (
+                    <div className="donation-edit-wrapper">
```

Comment:
need a `key={index}` to avoid warning (`Warning: Each child in an array or iterator should have a unique "key" prop.`)